csrc/scheduler/resize.cpp

```diff
@@ -0,0 +1,214 @@
+          [](IterDomain* logical_id) { return logical_id->isBroadcast(); })) {
+    return false;
+  }
+
+  return true;
+}
+
+std::unique_ptr<HeuristicParams> ResizeScheduler::computeHeuristics(
+    Fusion* fusion,
+    SchedulerRuntimeInfo& runtime_info,
+    HeuristicDataCache* data_cache) {
+  FUSER_PERF_SCOPE("ResizeScheduler::computeHeuristics");
+  auto params = std::make_unique<HeuristicParams>(SchedulerType::Resize);
+  params->cparams.index_type = runtime_info.getIndexType();
+  return params;
+}
+
+void ResizeScheduler::schedule(Fusion* fusion, const HeuristicParams* params) {
+  FUSER_PERF_SCOPE("ResizeScheduler::schedule");
+
+  FusionGuard fg(fusion);
+#if 0
+  fusion->printMath();
+  fusion->print();
+  std::cout << std::endl;
+#endif
+  scheduler_utils::clearMemorySpace(fusion);
+
+  scheduler_utils::cacheInputs(fusion, true);
+
+  for (auto expr : fusion->exprs()) {
+    if (!expr->isOneOf<SliceOp, PadOp>()) {
+      continue;
+    }
+
+    scheduler_tools::propagateResizeToInputs(expr);
+  }
+#if 0
+  fusion->print();
+  std::cout << std::endl;
+
+  for (auto tv : fusion->allTvs()) {
+    std::cerr << "scheduled T" << tv->name() << "\n";
+    if (tv->hasRoot()) {
+      std::cerr << "\tRoot: " << toDelimitedString(tv->getRootDomain()) << "\n";
+    }
+    std::cerr << "\tLogical: " << toDelimitedString(tv->getLogicalDomain())
+              << "\n";
+    std::cerr << "\tLoop: " << toDelimitedString(tv->getLoopDomain()) << "\n";
+    std::cerr << "\tAdditional ids: "
+              << toDelimitedString(tv->domain()->additionalIDs()) << "\n";
+    std::cerr << "\tInitial loop ids: "
+              << toDelimitedString(tv->domain()->initialLoop()) << "\n";
+    for (auto expr : tv->domain()->allExprs()) {
+      std::cerr << expr->toString(4);
+    }
+  }
+#endif
+  auto ref_tv = getReferenceTensor(fusion);
+
+  // Just simple scheduling for now.
+  // TODO: Do something smarter. Can just use the pointwise scheduler?
+
+  // Make sure the DID ID located at the outermost position
+  int64_t reorder_pos = 0;
+  std::unordered_map<int64_t, int64_t> old2new;
+  for (const auto i : c10::irange(ref_tv->getLoopDomain().size())) {
+    if (isParallelTypeDeviceDim(ref_tv->axis((int64_t)i)->getParallelType())) {
+      old2new.emplace((int64_t)i, reorder_pos);
+      ++reorder_pos;
+    }
+  }
+  ref_tv->reorder(old2new);
+
+  // Schedule only the remaining IDs
+  const auto outermost_pos = (int64_t)old2new.size();
+  ref_tv->flatten(outermost_pos);
+  ref_tv->split(outermost_pos, 128);
+  ref_tv->split(outermost_pos, 1 << 14);
+  ref_tv->axis(-1)->parallelize(ParallelType::TIDx);
+  ref_tv->axis(-2)->parallelize(ParallelType::BIDx);
+
+  // Propagate the reference to the other tensors
+  scheduler_tools::scheduleLoopDomainsLike(
+      fusion->allTvs(), ref_tv->getLoopDomain());
+
+  inlineMost();
+
+  // TODO: Alias support doesn't seem to be working. For example, see
+  // AliasTest.AliasOutputBeforeNonAliasOutput.
```

Comment:
Got it. That test verifies a performance optimization that the kernel doesn't need to write to an aliasing output. It doesn't affect functionality, so I'm fine with investigating it after the PR. 

$.02: can you try turn on https://github.com/NVIDIA/Fuser/blob/b53c103b22cd9febeecfe6ea0377649365382763/csrc/scheduler/mark_aliases.cpp#L30 to print more debugs? I suspect some allocation domains around the slice aren't in favor.

Reply:
```
[ RUN      ] AliasTest.AliasOutputBeforeNonAliasOutput
***Runtime***: Try to schedule fusion un-segmented:

Scheduler _expr_eval_ ***rejected*** because : Fusion must contain only a single expression.
Scheduler _no_op_ ***rejected*** because : output has a concrete dimension
Scheduler _matmul_ ***rejected*** because : No matmul patterns were found
Scheduler _reduction_ ***rejected*** because : No reduction op to schedule
***Accepted*** as: resize
[mark_aliases] Input fusion:
Inputs:
  T0_g_float[iS0{2}, iS1{3}]
Outputs:
  T1_g_float[iS12{1}, iblockIdx.x13{16384}, ithreadIdx.x11{128}] ca_pos( 3 )
  T2_g_float[iS14{1}, iblockIdx.x15{16384}, ithreadIdx.x16{128}] ca_pos( 3 ) produce_pos( 3 )

%kernel_math {
T1_g_float[iS12{1}, iblockIdx.x13{16384}, ithreadIdx.x11{128}] ca_pos( 3 )
   = slice( T0_g_float[iS0{2}, iS1{3}], { {0, 2, 1} {0, 2, 1} } )
T2_g_float[iS14{1}, iblockIdx.x15{16384}, ithreadIdx.x16{128}] ca_pos( 3 ) produce_pos( 3 )
   = T1_g_float[iS12{1}, iblockIdx.x13{16384}, ithreadIdx.x11{128}] ca_pos( 3 )
   + T1_g_float[iS12{1}, iblockIdx.x13{16384}, ithreadIdx.x11{128}] ca_pos( 3 );
} // %kernel_math

[mark_aliases] Alias analysis result:
  Potential aliases:
    T1 is an alias of T0 if its layout is <allocation=[iS2{2}, iS4{2}rf], contiguity=[f t]>
  Finalized aliases:
    T1 of allocation domain [iS2{2}, iS4{2}rf] and logical domain [iS2{2}, iS4{2}rf] is a transitive alias of T0

[mark_aliases] Marked T1 as an alias of T0
/home/nmaruyama/nvfuser/debug2/tests/cpp/test_alias.cpp:524: Skipped
Following aliss checks not supported yet

[  SKIPPED ] AliasTest.AliasOutputBeforeNonAliasOutput (335 ms)
[----------] 1 test from AliasTest (335 ms total)
```

Reply:
> [mark_aliases] Marked T1 as an alias of T0

I'll take a quick look. The above line means alias analysis did kick in and the resize scheduler indeed called aliasOutputToInput. I'm surprised that marked alias wasn't materialized in the CUDA kernel.

Reply:
Ha, I was just lazy and didn't bother with outputs yet. Didn't expect that would matter for the alias support. Thanks for taking a look!

Reply:
That indeed fixed the issue. Thanks!